src/Servers/IIS/AspNetCoreModuleV2/AspNetCore/applicationinfo.cpp

```diff
@@ -251,6 +251,26 @@ APPLICATION_INFO::ShutDownApplication(const bool fServerInitiated)
     }
 }
 
+/* Overview of shadow copy logic when enabled. See https://github.com/dotnet/aspnetcore/pull/28357 for more context
+
+ * On first request, ANCM goes through it's startup sequence, starting dotnet and sending the request into managed code. During this sequence, 
+ * ANCM will copy the contents of the app directory to another directory which is user specified. The path to this directory can be absolute or relative.
+
+ * Logs and log files will be written to the app directory rather than the shadow copy directory. app_offline will also only be watched in the app directory. 
+ * The current directory will be set to the app directory as well as the AppContext.BaseDirectory.
+
+ * On publish of new content to the app directory, ANCM will start debouncing file change notifications for dlls, waiting for a steady state. 
+ * This is done by resetting a timer each time a dll is changed, eventually triggering the timer once there are no dll changes. Afterwards, shutdown is started, 
+ * causing the process to recycle.
+
+ * Subfolders are created under the user specified shadowCopyDirectory, where the highest int value directory name will be used each time. 
```

Comment:
I assume that if directory '1' exists but '0' doesn't, '0' will be used and '1' will be deleted. Is that the case?

Reply:
That's a good question, I'll add a new shadowcopy test to see what happens, I know that it does go and clean up old directories, but I'm not sure if it will look for gaps like that, I'd sort of assume it would use '2' to keep increasing

Reply:
Yeah see https://github.com/dotnet/aspnetcore/pull/35249 for the new test so the behavior is '1' will be used, and then '2' will be used on the next modification, so its always incrementally increasing using the highest integer for the next location (copied on shutdown).